### ISSUE TITLE
keep for inner_join?

### ISSUE THREAD

**Issue by mmuurr** [tidyverse/dplyr#5893]
`left_join`, `right_join`, ... all support the `keep` option, with the exception of `inner_join`.
Is there a reason why?

An example where this would be useful is the ability to use additional verbs later in a pipeline without worrying about the exact order of upstream tables that were `inner_join`ed:
```r
x <- tibble(x_id = 1:20, some, other, data, here)
y <- tibble(y_id = 11:30, and, even, more, data, here)

inner_join(x, y, by = c("x_id" = "y_id"), keep = TRUE) %>%  ## `keep` is ignored
  filter(y_id %% 2 == 0)  ## oh noes, y_id is gone
```
Obviously one can do some `rename`-ing as part of a pipeline, but it's nice for owners of `x` and owners of `y` to now be able to write pipelines on the joined data while referring to the original columns most familiar to them (perhaps at the expense of some duplicated col data, though with careful copy-on-write implementation, even this can probably be avoided).

**Comment by dlindelof** [comment]
I believe this was implemented in #5581 and should be available starting from v1.0.3.

**Comment by DavisVaughan** [comment]
Yea, `keep` now exists for `inner_join()`

``` r
library(dplyr)

x <- tibble(x_id = 1:20)
y <- tibble(y_id = 11:30)

inner_join(x, y, by = c("x_id" = "y_id"), keep = TRUE) %>%  ## `keep` is ignored
  filter(y_id %% 2 == 0)
#> # A tibble: 5 x 2
#>    x_id  y_id
#>   <int> <int>
#> 1    12    12
#> 2    14    14
#> 3    16    16
#> 4    18    18
#> 5    20    20
```

<sup>Created on 2021-06-18 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
